content/blog/understanding-the-r-pipe.Rmd

```diff
@@ -0,0 +1,268 @@
+</center>
+
+## ACTUALLY I THINK WE NEED A DETOUR TO EXPLAIN ANONYMOUS (LAMBDA) FUNCTIONS BEFORE WE CAN EXPLAIN HOW `|>` WORKS
+
+Why anonymous? Because they are not named functions from a package or written by you and stored in a function object. Anonymous functions are created on-the-fly, applied immediately, and don't persist after they’ve been used: `function(x) {}`.
+
+```{r}
+function(x) {
+  x[which.max(x$mpg), ]
+}
+```
+
+What does this do? It creates an anonymous function (also known as a lambda function)
+ if it is not saved to an object.
+
+Introduced in R 4.1, the shortcut for anonymous functions `\(x) {}` is the same as `function(x) {}`:
+
+```{r}
+# equivalent to the above
+\(x) {
+  x[which.max(x$mpg), ]
+}
+```
+
+What you gain from writing an anonymous function is that you get to direct traffic by explicitly stating the inputs and how they will be used in the function. Going back to our discussion of pipes, you get to direct exactly where the LHS of the pipe goes in the RHS.
+
+## How the native R pipe works
+
+Like the {magrittr} pipe `%>%`, the native R pipe `|>` pipes the LHS into the first argument of the function on the RHS: `LHS |> RHS`.
+
+You can write:
+
+```{r}
+mtcars |> sqrt() |> head()
+```
+
+Which is equivalent to:
+
+```{r}
+head(sqrt(mtcars))
+```
+
+An important note: on the RHS of `|>`, you need to include the function as a function call, which means appending an `()` at the end of the function name, rather than just its name. For example, the square root function is called by writing `sqrt()`. If you try to run `mtcars |> sqrt` without the `()` at the end, you will get an error: `Error: The pipe operator requires a function call as RHS`.
+
+So, the native R pipe pipes the LHS into the first argument of the function on the RHS (with an extra requirement of needing a function call on the RHS). **But that’s all!** If you want to do anything _beyond_ piping the LHS into the first argument of the RHS function, then you need the special anonymous function syntax introduced above.
+
+
+A gotcha here is that we also needed to write parentheses around the anonymous function, such that a pseudocode version of the above is `mtcars |> (anonymous-function-definition)()`. The reason for this is so that the second set of `()` properly points to the complex expression inside the first set of `()` as the function being called.^[Curly braces for the first expression, `{}()`, works too: `mtcars |> {\(x) {x[which.max(x$mpg),]}}()`
+. See Ronak Shah's reply in this [StackOverflow thread](https://stackoverflow.com/questions/67633022/what-are-the-differences-between-rs-new-native-pipe-and-the-magrittr-pipe) for an example.] ^[Or more on this, see Q2 and its answer in the 'Functions' chapter of [Advanced R Solutions](https://advanced-r-solutions.rbind.io/functions.html).]
```

Comment:
Here's the direct link to the reply that you can use instead: https://stackoverflow.com/a/67638063

Reply:
Also on this line, change 'Or more on this' to 'Or for more on this'